### ISSUE TITLE
[java] java-unnecessarycode/UnnecessaryFinalModifier on try-with-resources

### ISSUE THREAD

**Issue by jsotuyod** [pmd/pmd#676]
When using try-with-resources, the resource variable is implicitly final. We should flag explicit modifiers as unnecessary.

**Code Sample demonstrating the issue:**

```java
try (final FileWriter fw = new FileWriter(new File("something.txt"))) {
  // do something on fw
}
```
